pkg/pgn/parser.go

```diff
@@ -31,7 +32,15 @@ func NewPGNParser(path string) (*PGNParser, error) {
 		file:       file,
 		totalBytes: bytesize.New(size)}
 
-	if strings.HasSuffix(path, "bz2") {
+	if strings.HasSuffix(path, "zst") {
+		pp.isArchived = true
+		zstReader, err := zstd.NewReader(file)
+		if err != nil {
+			return nil, fmt.Errorf("error opening zst: %s", err)
+		}
+		pp.zstReader = zstReader
+		pp.scanner = bufio.NewScanner(bufio.NewReader(zstReader))
+	} else if strings.HasSuffix(path, "bz2") {
```

Comment:
This is becoming messy and long. Probably should create a new function that takes the file/path and returns the Reader.
A cleaner code would be using:
`switch strings.HasSuffix(...) {
case "bz2":
...
case "zst":
...
default:
...
}
`